src/colorizer/ColorizeCanvas.ts

```diff
@@ -185,8 +252,66 @@ export default class ColorizeCanvas {
     this.setUniform("outlierColor", color);
   }
 
-  setHighlightedId(id: number): void {
-    this.setUniform("highlightedId", id);
+  setSelectedTrack(track: Track | null): void {
+    if (this.track && this.track?.trackId === track?.trackId) {
+      return;
+    }
+    this.track = track;
+    if (!track || !track.centroids || track.centroids.length === 0 || !this.frameResolution) {
+      return;
+    }
+    // Make a new array of the centroid positions,
+    // normalizing to screen space.
+    // We need three floats per coordinate
+    this.points = new Float32Array(track.centroids.length * 3);
+    for (let i = 0; i < track.centroids.length; i++) {
+      // Screen space is in [-1, 1] range.
+      this.points[3 * i + 0] = (track.centroids[i][0] / (4.0 * this.frameResolution.x)) * 2.0 - 1.0;
+      this.points[3 * i + 1] = -((track.centroids[i][1] / (4.0 * this.frameResolution.y)) * 2.0 - 1.0);
```

Comment:
something about this math here seems to be fighting against what is in the vertex shader.  I believe you could just pass the values as centroids in their pixel coordinates and have the shader do the rest of the transformation from pixels to (-1,1) range by dividiing by frame textureSize

Reply:
alternatively, if you transform the coordinates to the -1..1 space HERE, then you shouldn't need to transform the positions at all in the vertex shader?

Reply:
So, the canvas has coordinates in [-1, 1], but the way the image is resized in the canvas element makes the vertex coordinates not map correctly. The logic in the vertex shader scales it in the same way that the canvas is scaled.

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/142cf751-ca0f-40f0-a468-1f0e2842eaca)
![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/38d24e5b-5970-4aaf-b01d-e4ca225a3ed3)

I agree that passing pixel coords in as the vertex coordinates makes more sense, though, and I'll update the logic accordingly!

Reply:
Yup I was totally missing the resizing aspect ratio behavior.  But yeah I still agree with just passing the coordinates in as pixels.

Reply:
could something like this be accomplished by setting the line's `scale`?

Reply:
yep that's another possibility - if you use the threejs `scale` then you are integrating more closely with threejs infrastructure and then it would be natural to use the modelview transform in the vertex shader.